src/servers/database.ts

```diff
@@ -697,7 +697,9 @@ export class DatabaseServer extends SkeletonServer {
         }
         return bytesToHex(transactionHash);
       },
-      err: (code) => {
+      err: (message) => {
+        const sepIndex = message.indexOf('|')
+        const code = sepIndex > -1 ? message.substring(0,19) : message
```

Comment:
This seems dubious: why index `19` instead of `sepIndex`?

Also, we might need to consider testing for an HTML response prior to doing this split, as HTML error responses (from proxies, etc) could certainly include the pipe character. On the other hand, we would want the gas burned for HTML error responses as well, meaning that perhaps we ought to be looking for and splitting on the _last_ pipe character in the message.

Reply:
Should we pass html error in the header? If so - we will need to sanitise html body, since there could be unallowed characters([rfc7230](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.6))

Reply:
In general we shouldn't pass back the HTML body, that's been a wishlist item that we would detect 502s and the like (which usually come with HTML bodies) and just give a one-line error message back.